---
title: "02CLUS_Fig2_H3k4me3_CORR_k6"
author: "Dr. Julio Domingo Cordero Henriquez ; email=julio.cordero@medma.uni-heidelberg.de"
paper: "Zebrafish heart Regeneration"
output: html_document
---
<style type="text/css">
body{ /* Normal  */
      font-size: 12px;
  }
td {  /* Table  */
  font-size: 8px;
}
h1.title {
  font-size: 38px;
  color: black;
}
h1 { /* Header 1 */
  font-size: 28px;
  color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 22px;
  color: black;
}
h3 { /* Header 3 */
  font-size: 18px;
  font-family: "Arial", Times, serif;
  color: black;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
# **General Information of this Script**
```c
                                    ## **Aim of the Script** 
##### To cluster the enrichment of H3K4me3 histone modification during the zebrafish heart regeneration.

## **Input files**
### 02RE_analysis_PRJNA509429_CLUS.csv
#### -RNA_seq matrix from PRJNA509429. They performed RiboZero-RNA-seq in the same time points that the ChIP-seq

### 01ANNO_PEAKS_H3K4me3_ZF_ALL_x_BEDTO.bed
#### -All the H3K4me3 peaks were summary together in one matrix with their respective annotations 

### 02MAT_NORMA_CLUS_Fig2_H3k4me3_CORR.txt
#### - Normalized zscore mean RPKM counts from H3K4me3 during zebrafish heart regeneration

## **Output files**
#### 001_Fig2C_H3k4me3_CORR_k6_Fig2C_Spearman_rank.pdf (Fig2C)
####- Bar plot showing the number of correlated genes.

#### 002_002_for_TABLE2_02CLUS_Fig2_H3k4me3_CORR.xls
####- Matrix of the annotated peaks containing the clusters from H3K4me3.

#### 003_FIG2F_02CLUS_Fig2_H3k4me3_CORR_k6_CLUS_LINEs.pdf
####- Line plot containing the enrichment of the H4K4me3 enrichment.

#### 005_forFigS2_02CLUS_Fig2_H3k4me3_CORR_k6_by_CLUS.pdf
####- Bar plot showing the correlation by clusters.

#### 00forMOTIFs
 ####- Folder containing all the peak files to be used for motif search

```

## Programs to load in R
```{r,message=FALSE,echo=FALSE,warning=FALSE}

library(openxlsx)
library(data.table)
library(edgeR)
library(dplyr)
library(corrplot)
library(doParallel)
library(BiocParallel)
library(gam)
library(foreach)
library(ggplot2)
library("RcppArmadillo")
library(DESeq2)
library(tidyr)
library(RColorBrewer)
library(Rcpp)
library(clusterProfiler)
library(org.Dr.eg.db)
library(plyr)
library(gplots)
library(org.Rn.eg.db)
library(EnrichedHeatmap)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(RColorBrewer)
library(future)
plan()
plan("sequential", workers = 8)
options(future.globals.maxSize = 100000 * 1024^2)
```


### **Set the working environment and Folder to output the results**
```{r,class.source="bg-info",echo=TRUE,warning=FALSE}
 workdir = "./"
setwd(workdir)
PTHA1="../03OUPUT/"
dir.create (PTHA1)
PTHA=paste(PTHA1,"02CLUS_Fig2_H3k4me3_CORR_k6/",sep="")
PORT=paste(PTHA1,"02CLUS_Fig2_H3k4me3_CORR_k6/","00forNGS/",sep="")
PORT2=paste(PTHA1,"02CLUS_Fig2_H3k4me3_CORR_k6/","00forMOTIFs/",sep="")
PORT3=paste(PTHA1,"02CLUS_Fig2_H3k4me3_CORR_k6/","00GOs/",sep="")
dir.create (PTHA)
dir.create (PORT)
dir.create (PORT2)
dir.create (PORT3)
PROJECT="02CLUS_Fig2_H3k4me3_CORR_k6"
PROJECT2="02CLUS_Fig2_H3k4me3_CORR_k6"
#### FOR VOLCANO
upcol<- "#8E0152" # magenta from PiyG
nc<- "#000000" # black
#downcol<- "#7fbc41" # green from PiyG
downcol<- "#4c6ced" # green from PiyG
CC= c(downcol, "lightgray",upcol)
PEAK="Peak group"
 NAME33b<-c("00", "01", "04", "14", "45") 
```

###### Load RNA-seq from 02RE_analysis_PRJNA509429_CLUS #########
```{r,class.source="bg-info",echo=TRUE,warning=FALSE}
meta = "../../../00MASTER_Input_FILES/01RNA_seq_ZF_PRJNA509429/02RE_analysis_PRJNA509429_CLUS.csv"
mat_m = read.delim(meta,header=T,check.names=FALSE, stringsAsFactors=FALSE)
mat_m$SYMBOL<- mat_m$zfin_id_symbol
mat_m$Clu_EXP <- mat_m$Clu
NAME11<- data.frame(colnames(c(mat_m[,2:5],mat_m[,7])))
NAME11b<- data.frame(colnames(mat_m))
mat_m<- mat_m[2:nrow(mat_m),]

######## select Log2FC
mat_m1a<- data.frame(NAME11b[grep("log2fc_*", NAME11b[,1]), ])
mat_m1a<- t(mat_m1a)
mat_m11<- data.frame(NAME11b[grep("mean_*", NAME11b[,1]), ])
mat_m1b<- t(mat_m11[1:4,])
mat_m1b12<- t(mat_m11[6,])
mat_m1c <- cbind(mat_m1b,mat_m1b12)
name5=c("EMSEMBL", "rgd_symbol","ensembl_gene_id","length",mat_m1c)

NAME_Z<- paste("z",mat_m1c,sep="")
mat_m3<- subset(mat_m,select=c("SYMBOL","Clu_EXP",mat_m1c))
mat_m34 = t(scale(t(mat_m3[,3:ncol(mat_m3)])))
colnames(mat_m34)<- NAME_Z
mat_m33 <- cbind(mat_m3,mat_m34)
```


### Load the H3K4me3 peak files

```{r,message=FALSE,class.source="bg-info",warning=FALSE}
############ Peaks from H3k4me3 total  #######################
NAME12<- c(1,2,3,4,5,6,7,8,9,10,11,12,15,17)
NAME00<- paste("V",NAME12,sep="")
NAME01<- c("chr",	"start",	"end","chrP",	"startP",	"endP",	"PEAK_ID",	"ID" ,"SYMBOL", "EMS",	"length",	"CLU",	"Distance.to.TSS",	"CLU2")
PEAKS=unzip("../01DATA/01ANNO_PEAKS_H3K4me3_ZF_ALL_x_BEDTO.zip")
mat_PEAKS2 = read.delim(PEAKS,header=F,check.names=FALSE, stringsAsFactors=FALSE,sep="\t")
NAME11b<- data.frame(colnames(mat_PEAKS2))
NAME11b<- t(NAME11b)

mat_PEAKS2_1<- subset(mat_PEAKS2,select=NAME00)
colnames(mat_PEAKS2_1)<- NAME01
mat_PEAKS2_1$PEAK_ID2<- paste(mat_PEAKS2_1$chr,mat_PEAKS2_1$start,mat_PEAKS2_1$end,sep="_")
#length(unique(mat_PEAKS2_1$PEAK_ID2))

```

### Load the count matrix
```{r,message=FALSE,class.source="bg-info",warning=FALSE}
adD1 = unzip("../01DATA/02MAT_NORMA_CLUS_Fig2_H3k4me3_CORR.zip")
cat("Current file name is:",adD1,"/n")
mat_adD = read.table(adD1,header=T,check.names=FALSE, stringsAsFactors=FALSE,sep="\t")
NAME11c<- data.frame(colnames(mat_adD))
mat_adD$Clu_HIS<- mat_adD$Clu
mat_adDu <- mat_adD[!duplicated(mat_adD$PEAK_ID2),]
mat_P<- data.frame(NAME11c[grep("mean_*", NAME11c[,1]), ])
mat_P<- t(mat_P)
length(unique(mat_adD$SYMBOL))

#table(mat_adD$CLU)
##########   MERGE  with RNA_SEQ
mat_adD1<- subset(mat_adD,select=c("PEAK_ID2","SYMBOL","ID", "CLU","Clu","CLU2",mat_P))

NAME_P<- colnames(mat_P)<- paste("H",mat_P[1,],sep="_")
colnames(mat_adD1)<- c("PEAK_ID2","SYMBOL","ID", "CLU","Clu_HIS","CLU2",NAME_P)
MI=length(unique(mat_adD$Clu))
#length(unique(mat_adD1$PEAK_ID2))
#length(unique(mat_adD1$SYMBOL))
```


### To check the Broadness first to check if the files used are correct
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE,fig.keep='none'}
MA6<- data.frame(table(mat_adD1$CLU,mat_adD1$CLU2,mat_adD1$ID))
XL<- "dpci"
YL="zscoreLog2_meanRPKM+1"
library(gridExtra)
BASIC_COL="black"
YL_EXP="Log2(meanRPKM+1)"
IP="H3k4m3"
YL_EXP2="Zscore"
LE=14
SI=14
PEAK="Peak group"
CO22P <-  c("#4d004b","#9ebcda", "gray","darkblue", "black")
p11 <- ggplot(MA6, aes(factor(Var3), Freq, fill = factor(Var1))) +  geom_bar(position = "fill",stat = "identity") +   guides(fill = guide_legend(reverse = F)) +labs(fill = PEAK)+theme_bw()

p61 <- p11 +   scale_fill_manual(values =CO22P) +  theme(strip.placement = "inside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0)) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL)+ scale_x_discrete(labels=NAME33b)
p6a1 <- p11 +   scale_fill_manual(values =CO22P) +  theme(strip.placement = "inside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0)) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL)+ scale_x_discrete(labels=NAME33b)+
    facet_wrap(~Var2, ncol = 4,strip.position="top",scales="fixed")

print(p61)
print(p6a1)
```

### MERGE CLUS from HISTONE to EXPRESSION
```{r,message=FALSE,class.source="bg-info",warning=FALSE}
##########   MERGE  with RNA_SEQ
MA1 <- list(mat_m33,mat_adD1) %>%
  Reduce(function(dtf1,dtf2) inner_join(dtf1,dtf2,by="SYMBOL"), .)
#length(unique(MA1$SYMBOL))
#length(unique(MA1$CLU))
#MA1u <- MA1[!duplicated(MA1$PEAK_ID2),]
MA2 <- MA1[!duplicated(MA1$SYMBOL),]

MA1u<- MA1[!duplicated(MA1[c("SYMBOL","CLU")]),] 
 NAME12_MA2<- data.frame(colnames(MA2))
NAME11d<- data.frame(colnames(MA1u))

#length(unique(MA1u$SYMBOL))
#length(unique(MA1u$PEAK_ID2))
#length(unique(MA1u$CLU))
##extract name for the histone enrichment
mat_m1b2<- data.frame(NAME11d[grep("H_zRPKMmean*", NAME11d[,1]), ])
mat_m1b2<- t(mat_m1b2)
mat_m1b3<- data.frame(NAME11d[grep("zmean*", NAME11d[,1]), ])
mat_m1b3<- t(mat_m1b3)
NAME55<- c("00","01","04","14","45")
NAME55b<- c("01","04","14","45")
mat_adD1b<- subset(MA2,select=c("SYMBOL","ID", mat_m1b2, "Clu_HIS","CLU","CLU2","Clu_EXP"))

colnames(mat_adD1b)=c("SYMBOL","ID", NAME55, "Clu_HIS","CLU","CLU2","Clu_EXP")

mat_RPKM1_5_c <- gather(mat_adD1b, NAME55, key="Sample",value= "log2fc")
mat_RPKM1_5_c$log2fc<- log2(mat_RPKM1_5_c$log2fc+1)
CO<-brewer.pal(n = 8, name = "Dark2")

mat_RPKM1_5_c2<- replace_na(mat_RPKM1_5_c,list(CLU="TO_FIND"))

CHECK_CLU2<- data.frame(unique(mat_RPKM1_5_c$CLU2))
CHECK_CLU<- data.frame(unique(mat_RPKM1_5_c$CLU))

#table(mat_RPKM1_5_c$Clu_EXP)
### Names ##
library(ggpubr)
XL<- "dpci"
YL_EXP="Log2(meanRPKM+1)"
IP="H3k4m3"
#rm(p)
YL_EXP2="Zscore"
BASIC_COL="black"
LE=12
```


## CORRELATION Analysis with Spearkman correlation
```{r,message=FALSE,class.source="bg-info",warning=FALSE}
NAME11d<- data.frame(colnames(MA1u))
##extract name for the histone enrichment
mat_m1b2<- data.frame(NAME11d[grep("H_zRPKMmean*", NAME11d[,1]), ])
mat_m1b2<- t(mat_m1b2)
mat_m1b3<- data.frame(NAME11d[grep("zmean*", NAME11d[,1]), ])
mat_m1b3<- t(mat_m1b3)
NAME33<- data.frame(table(MA1$CLU))
MAT12_t<- MA1u
DIS_list<- list()
DIS_list1<- list()
for (i in 1:nrow(MAT12_t)){  
x<-MAT12_t[i, mat_m1b2]
y<-MAT12_t[i, mat_m1b3]
results<-cor.test(as.numeric(x),as.numeric(y), method = "spearman")
results$CLU<- MAT12_t[i, "CLU"] 
DIS_list[[i]] <- results$p.value
DIS_list1[[i]]<- results$estimate
DIS_list[[i]]$CLU<- MAT12_t[i, "CLU"] 
}
pvalue_spear<- do.call(rbind, DIS_list)
GN5b<- do.call(rbind, DIS_list1)
#length(pvalue_spear)
#length(GN5b)
MAT12_t2<- cbind(MAT12_t,pvalue_spear,GN5b)
#summary(MAT12_t2$COR)
```

##### CORRELATION to extract the Cut-off of the Analysis with Spearkman correlation

```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
NAME_BRO<- data.frame(table(MAT12_t2$CLU))
FIL<- data.matrix(summary(MAT12_t2$rho))
FIL2<-as.numeric(round(FIL[2,1],digits = 2))
FIL5<-as.numeric(round(FIL[5,1],digits = 2))
FIL3<- FIL2*-1

MAT12_t2$COR<- as.numeric(MAT12_t2$rho)
MAT12_t2$PV_cor<- as.numeric(MAT12_t2$V1)
PV_cor<- data.matrix(summary(MAT12_t2$PV_cor))
PV_cor2<-as.numeric(round(PV_cor [2,1],digits = 2))
PV_cor_log10<- log10(PV_cor2)*(-1)
MAT12_t2$Log10P<- log10(MAT12_t2$PV_cor)*(-1)
FIL25<- data.frame(rbind(FIL2,FIL3))
DIST1<- list()
DIST2<- list()
for(A in 1:length(FIL25[,1])) {
for(i in 1:length(NAME_BRO[,1])) {
#print(i)
#print(A)
DIST1[[i]]<- subset(MAT12_t2,CLU==NAME_BRO[i,1] & rho>=FIL3 & PV_cor<PV_cor2 )
DIST2[[i]]<- subset(MAT12_t2,CLU==NAME_BRO[i,1] & rho<= FIL2 & PV_cor<PV_cor2)
DIST1[[i]]$COM<- "01POS"
DIST2[[i]]$COM<- "02NEG"
}
}
CHECK23_t <- subset(MAT12_t2,CLU=" " )
CHECK23_t$COM<- "00TOTAL"
GN5_1<- do.call(rbind, DIST1)
GN5_2<- do.call(rbind, DIST2)
GN5b2<- rbind(GN5_1,GN5_2,CHECK23_t)
#summary(MAT12_t2$PV_cor)
C1<- plyr::count(GN5b2, c("CLU", "COM"))
C1GN5_1<-  plyr::count(GN5_1, c("CLU", "COM"))

CHECK22_bro<- subset(MAT12_t2,CLU=="01BROAD" & rho>=FIL3 & PV_cor<PV_cor2 )
CHECK24_nar<- subset(MAT12_t2,CLU=="03NARROW" & rho>=FIL3 & PV_cor<PV_cor2 )

CHECK22_bro2<- subset(MAT12_t2,CLU=="01BROAD" & rho<=FIL2 & PV_cor<PV_cor2 )
CHECK24_nar2<- subset(MAT12_t2,CLU=="03NARROW" & rho<=FIL2 & PV_cor<PV_cor2)

B1<- data.frame(table(CHECK22_bro$CLU))
B3<- data.frame(table(CHECK24_nar$CLU))
B11<- data.frame(table(CHECK22_bro2$CLU))
B31<- data.frame(table(CHECK24_nar2$CLU))
B_T<- rbind(B1,B3,B11,B31)

BR<- merge(CHECK22_bro,CHECK22_bro2,by="SYMBOL")
```

## normalize by the total number of genes in each group
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
NAME33b<- c("01BRO","03NAR")
GN5c<- rbind(GN5_1,GN5_2)
GN5c$CLU[GN5c$CLU=="02MEDIUM"] <- "01BROAD"
MA7<- plyr::count(GN5c, c("CLU", "COM"))
#length(unique(GN5c$SYMBOL))
 GN5cu<- GN5c[!duplicated(GN5c[c("SYMBOL")]),] 
MA7u<- plyr::count(GN5cu, c("CLU", "COM"))
MA7_t<- plyr::count(CHECK23_t, c("CLU", "COM"))
CO22_COR <-  c("darkblue","grey","gold","#4393C3","brown", "black")
XL="H3k4me3 peak group"
YL="Peak distribut (%)"
YL2="number of Peaks"
BASIC_COL="black"
SI=14
LE=14
GEN_C<- sum(MA7$freq)
MA7_p<- subset(MA7,COM=="01POS")
MA7_n<- subset(MA7,COM=="02NEG")
GEN_p<- sum(MA7_p$freq)
GEN_n<- sum(MA7_n$freq)
```

```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
GEN_C1<- sum(MA7u$freq)
MA7_p<- subset(MA7u,COM=="01POS")
MA7_n<- subset(MA7u,COM=="02NEG")
GEN_p1<- sum(MA7_p$freq)
GEN_n1<- sum(MA7_n$freq)
TRY="_BY_SMBOL_"
pdf(file=paste(PTHA,"001_",TRY,PROJECT,"_Fig2C_","Spearman_rank",".pdf", sep=""),width=5, height=4)  
P3s<-  ggplot(MA7u, aes(factor(CLU), freq, fill = factor(COM))) +geom_bar(position = "stack",stat = "identity") + guides(fill = guide_legend(reverse = F)) +
    labs(fill = "corr_Spearman's_rank") + theme_bw() + scale_fill_manual(values =CO22_COR) +    theme(plot.title= element_text(color=BASIC_COL, size=14, face="bold.italic"), axis.text.y= element_text(size=LE, face="bold"),axis.text.x = element_text(size=LE, face="bold"), axis.title.x = element_text(color=BASIC_COL, size=14, face="bold"), axis.title.y = element_text(color=BASIC_COL, size=14, face="bold"), panel.border = element_rect(colour = BASIC_COL, fill=NA, size=2))+labs(title=paste("Fig2C", GEN_p1,"peaks_pos",GEN_n1, "peaks_neg",GEN_C1,"Spea_rank", sep="_" ) , x=XL, y = "N° of genes")+ scale_x_discrete(labels=(NAME33b))
print(P3s)

dev.off()
```


#########   Number of genes correlated with the H3K4me3 ChIP-seq  #########
```{r,fig.dim = c(5, 5),class.source="bg-info"}
print(P3s)
```



```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
library(tidyr)
mat_RPKM1_5_c <- subset(GN5c,select=c("CLU","COR","Log10P"))
mat_RPKM1_5_c2 <- subset(GN5cu,select=c("CLU","COR","Log10P"))
```

## export differentially Correlated matrix
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
#write.xlsx(GN5c,file=paste(PTHA, PROJECT,"DIFF_CORR_SPEAR",".xlsx",sep=""),overwrite = T)
#write.xlsx(GN5cu,file=paste(PTHA,PROJECT,"DIFF_CORR_SPEAR_UNIQUE",".xlsx",sep=""),overwrite = T)
```

## to check the correlation 

```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
library(corrplot)
#CORR_MA <- subset(GN5c,select=c(mat_m1b2[1,3],mat_m1b3))
#M_s<-cor(CORR_MA, method="spearman")
#col<- colorRampPalette(c("deeppink", "white", "blue"))(20)
#corrplot(M_s, method="pie",type="upper",col=col)

### Matrix for clusters
mes_cp_cm_quasy_u2_esc2<- subset(GN5c,select=c("PEAK_ID2", "SYMBOL","CLU","COR","Log10P", mat_m1b2) )
name4z<- mat_m1b2
#length(unique(mes_cp_cm_quasy_u2_esc2$PEAK_ID2))
#length(unique(mes_cp_cm_quasy_u2_esc2$SYMBOL))
#write.table(mes_cp_cm_quasy_u2_esc2, file=paste(PTHA,PROJECT,TRY,"mes_cp_cm_quasy_u2_esc2_to_CHECK.txt",sep=""), dec=".",sep="\t",col.names=T,row.names=F,quote = F)

```


################# CONNECTION WITH THE CLUSTER!!!!  ###############
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE,fig.keep='none'}

########################Determine number of Clusters Example 2#########################
wss <- (nrow(mes_cp_cm_quasy_u2_esc2[,name4z])-1)*sum(apply(mes_cp_cm_quasy_u2_esc2[,name4z],2,var))
for (i in 2:15) wss[i] <- sum(kmeans(mes_cp_cm_quasy_u2_esc2[,name4z], centers=i)$withinss)
{par(lwd=3)
plot(1:15, wss, type="b", lwd =3,xlab="Number of Clusters",
     ylab="Within groups sum of squares",main=paste(PROJECT, "mat_all","to know how many 
     Cluster the data could have")) 
}

{pdf(file=paste(PTHA,"002_FigS2D_",PROJECT,"_ELBOWL_PLOT.pdf",sep="")  )
par(lwd=3)
plot(1:15, wss, type="b", lwd =3,xlab="Number of Clusters",
     ylab="Within groups sum of squares",main=paste(PROJECT, "mat_all","to know how many 
     Cluster the data could have")) 
}
dev.off()

```


```{r,fig.dim = c(5, 5),class.source="bg-info"}
plot(1:15, wss, type="b",lwd =3,xlab="Number of Clusters",   ylab="Within groups sum of squares",main=paste(PROJECT, "mat_all","to know how many 
     Cluster the data could have")) 
```


## K-MEANS unsupervised clustering 
```{r,class.source="bg-info",echo=TRUE,warning=FALSE,echo=FALSE}
##  Use set.seed to set a seed for the random value before doing the clustering.
MI=6
## VERY IPORTANT to get the same results
#library(NbClust)
CHECKF<- is.na(mes_cp_cm_quasy_u2_esc2[,name4z])
CHECKF<- subset(CHECKF,)
#head(mes_cp_cm_quasy_u2_esc2[,name4z])

set.seed(1)
clusters <- kmeans(mes_cp_cm_quasy_u2_esc2[,name4z],MI,iter.max=500,algorithm = "Hartigan-Wong") 
set.seed(1)
clusters2 <- kmeans(mes_cp_cm_quasy_u2_esc2[,name4z],MI,iter.max=500,algorithm = "Hartigan-Wong")

identical(clusters,clusters2)

mes_cp_cm_quasy_u2_esc2$Clu <- as.factor(clusters$cluster)
#str(clusters2)
str(clusters)
NAME55<- c("00","01","04","14","45")
#head(mes_cp_cm_quasy_u2_esc2)
mes_cp_cm_quasy_u2_esc22<- subset(mes_cp_cm_quasy_u2_esc2,select=c("PEAK_ID2",name4z,"Clu"))

colnames(mes_cp_cm_quasy_u2_esc22)=c("PEAK_ID2", NAME55,"Clu")

mat_RPKM1_5_c <- gather(mes_cp_cm_quasy_u2_esc22, NAME55, key="Sample",value= "log2fc")

#write.xlsx(mes_cp_cm_quasy_u2_esc22,file=paste(PTHA, PROJECT,"mes_cp_cm_quasy_u2_esc22",".xlsx",sep=""),overwrite = T)
```
##### to perform the cluster analysis only from the DIFF correlated genes to peaks
```{r,message=FALSE,class.source="bg-info",warning=FALSE}
data_summary2 <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,varname)
  data_sum <- rename(data_sum, c("mean" = varname))
  return(data_sum)}

#######################################
df2 <- data_summary2(mat_RPKM1_5_c, varname="log2fc", groupnames=c("Sample", "Clu"))
# Convert dose to a factor variable
df2$Sample=as.factor(df2$Sample)
library(RColorBrewer)
CO<-brewer.pal(n = MI, name = "Dark2")
CO1<-brewer.pal(n = MI, name = "Set2")
#####################
XL<- "dpci"
YL="zscoreLog2_meanRPKM+1"
library(gridExtra)
BASIC_COL="black"
YL_EXP="Log2(meanRPKM+1)"
IP="H3k4m3"
YL_EXP2="Zscore"
LE=14
SI=14
PEAK="Peak group"
```


```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
XL<- "dpci"
YL="zscoreLog2_meanRPKM+1"
NAME55<- c("00","01","04","14","45")
## THANK YOU JESUS!!
p11a<- ggplot(df2, aes(x=Sample, y=log2fc, group= as.factor(Clu),colour=as.factor(Clu))) +geom_line(size=2) + geom_point() + facet_wrap(~Clu, ncol = MI/2) + geom_smooth(method = 'gam') +theme_classic() +scale_color_manual(name="H3K4me3_CLUs",values=CO) +labs(title=paste("003_FIG2F_", PROJECT,sep="" ) , x=XL, y = YL)+theme(plot.title = element_text(color=BASIC_COL, size=SI, face="bold.italic"),axis.title.x = element_text(color=BASIC_COL, size=SI, face="bold"),axis.title.y = element_text(color=BASIC_COL, size=SI, face="bold"),axis.text.x=element_text(angle = 0, size=SI, face="bold"),axis.text.y=element_text(angle = 0, size=SI, face="bold"))

p11b<- p11a+theme_bw(base_size = 10)+ theme(plot.title = element_text(color=BASIC_COL, size=SI, face="bold.italic"),axis.title.x = element_text(color=BASIC_COL, size=SI, face="bold"),axis.title.y = element_text(color=BASIC_COL, size=SI, face="bold"),axis.text.x=element_text(angle = 0, size=SI, face="bold"),axis.text.y=element_text(angle = 0, size=SI, face="bold"),axis.title = element_text(angle = 0, size=SI, face="bold"))

pdf(file=paste(PTHA,"003_FIG2F_", PROJECT, "_CLUS_LINEs.pdf",sep=""),width = 3.5,height = 7) 
arrange0b <- ggarrange(p11b,    ncol = 1,nrow =1,common.legend = T, align = c("hv"),legend="right")
arrange0b1 <- ggarrange( p11a,    ncol = 1,nrow =1,common.legend = T, align = c("hv"),legend="right")

print(arrange0b) 
print(arrange0b1) 
dev.off()
print(arrange0b) 
print(arrange0b1) 
```


```{r,fig.dim = c(7, 8),class.source="bg-info"}
#print(arrange0b)+ print(arrange0b1)
```

### Print the matrixes for the Histone enrichment 
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
NAME88b<- c("chrP","startP","endP", "ID", "PEAK_ID",	"Distance.to.TSS",	"CLU2","ID")
GN5c22<- subset(GN5c,select=c("SYMBOL","COM"))
GN5c22_2 <- list(GN5c22,mes_cp_cm_quasy_u2_esc2) %>%
Reduce(function(dtf1,dtf2) inner_join(dtf1,dtf2,by="SYMBOL"), .)

mat_PEAKS2_2<- subset(mat_PEAKS2_1,select=c("PEAK_ID2",NAME88b))

MAT11 <- list(mat_PEAKS2_2,GN5c22_2) %>%
Reduce(function(dtf1,dtf2) inner_join(dtf1,dtf2,by="PEAK_ID2"), .)
MAT12 <- list(mat_m33,MAT11) %>%
Reduce(function(dtf1,dtf2) inner_join(dtf1,dtf2,by="SYMBOL"), .)
CHECK2<- subset(MAT11,SYMBOL=="cdh2")

MAT11u<- MAT11[!duplicated(MAT11[c("PEAK_ID2","chrP","startP","endP")]),] 
#write.table(mes_cp_cm_quasy_u2_esc22, file=paste(PTHA,PROJECT,TRY,"Matrix_ZSCORE.csv",sep=""), dec=".",sep="\t",col.names=T,row.names=F,quote = F)
#write.table(MAT12, file=paste(PTHA,PROJECT,TRY,"Matrix_ALL.csv",sep=""), dec=".",sep="\t",col.names=T,row.names=F,quote = F)
#write.xlsx(MAT12,file=paste(PTHA,PROJECT,"_ALL2",".xlsx",sep=""), overwrite = T)
write.xlsx(MAT11u,file=paste(PTHA,"002_for_TABLE2_",PROJECT,".xlsx",sep=""), overwrite = T)
```

#### for MOTIFS
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=T}
NAME_ENH<- data.frame(table(MAT11u$Clu))
for(i in 1:length(NAME_ENH$Var1)) {
#print(i)
MA499u1<- subset(MAT11u,Clu==NAME_ENH[i,1],select=c("chrP","startP","endP","PEAK_ID2","Clu"))
write.table(MA499u1,file=paste(PORT2, PROJECT, "Clu",NAME_ENH[i,1],"_forMOTIF.bed",sep=""),sep="\t",row.names = F,col.names=F,dec=".",quote = F)
}
```

### for NGSPLot EMSEMBL; it was not used in this manuscript. However, the script produce these files to potentially used for other experiments
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
G1 <- bitr(mes_cp_cm_quasy_u2_esc2$SYMBOL, fromType="SYMBOL", toType=c("ENSEMBL"), OrgDb="org.Dr.eg.db")
G2<- merge(G1, mes_cp_cm_quasy_u2_esc2, by="SYMBOL")
NAME_ENH<- data.frame(table(G2$ENH4))
for(i in 1:length(G2$Var1)) {
#print(i)
MA499u2<- subset(G2, Clu==NAME_ENH[i,1])
write.table(MA499u2$ENSEMBL,file=paste(PORT,PROJECT,"CLu",i,".txt",sep=""),sep="\t",row.names = F,col.names=F,dec=".",quote = F)
}
```

## GO analysis by cluster
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE,fig.keep='none'}
differ_2 = subset(mes_cp_cm_quasy_u2_esc2,  select= c("SYMBOL","Clu"))
#head (differ_2)
colnames(differ_2)= c("SYMBOL","clu")
#length(unique(mes_cp_cm_quasy_u2_esc2$SYMBOL))
differ_3 <- bitr(mes_cp_cm_quasy_u2_esc2$SYMBOL, fromType="SYMBOL", toType=c("ENTREZID"), OrgDb="org.Dr.eg.db")
differ_4 = merge(differ_3,differ_2, by= "SYMBOL")
differ_4_u <- differ_4[!duplicated(differ_4$SYMBOL),]
colnames(differ_4)= c("SYMBOL", "Entrez","clu")
differ_4_1 = subset(differ_4, clu=="1")
differ_4_2 = subset(differ_4, clu=="2")
differ_4_3 = subset(differ_4, clu=="3")
differ_4_4 = subset(differ_4, clu=="4")
differ_4_5 = subset(differ_4, clu=="5")
differ_4_6 = subset(differ_4, clu=="6")

formula_resddBGI11FC1_25_1_5_bonfe <- compareCluster(Entrez~clu, data=differ_4,fun = "enrichGO",OrgDb="org.Dr.eg.db", ont= "BP", pAdjustMethod= "bonferroni",pvalueCutoff  = 0.05,qvalueCutoff  = 0.05,readable = T)

GO00<- dotplot(formula_resddBGI11FC1_25_1_5_bonfe,showCategory=5, title=paste(PROJECT, TRY,"_bonferro_test",sep=""))

formula_resddBGI11FC1_25_1_5_FDR <- compareCluster(Entrez~clu, data=differ_4,fun = "enrichGO",OrgDb="org.Dr.eg.db", ont = "BP",pAdjustMethod = "fdr", pvalueCutoff  = 0.05,qvalueCutoff  = 0.1, readable = T)
GO001<- GO00+ theme(plot.title = element_text(color=BASIC_COL, size=SI, face="bold.italic"),axis.title.x = element_text(color=BASIC_COL, size=SI, face="bold"),axis.title.y = element_text(color=BASIC_COL, size=SI, face="bold"),axis.text.x=element_text(angle = 0, size=SI, face="bold"),axis.text.y=element_text(angle = 0, size=SI, face="bold"))

GO00b<- dotplot(formula_resddBGI11FC1_25_1_5_FDR,showCategory=5, title=paste(PROJECT, TRY,"_FDR_test",sep=""))

GO00b1<- GO00b+ theme(plot.title = element_text(color=BASIC_COL, size=SI, face="bold.italic"),axis.title.x = element_text(color=BASIC_COL, size=SI, face="bold"),axis.title.y = element_text(color=BASIC_COL, size=SI, face="bold"),axis.text.x=element_text(angle = 0, size=SI, face="bold"),axis.text.y=element_text(angle = 0, size=SI, face="bold"))


ewp_1 <- clusterProfiler::enrichGO(gene=differ_4_1[[2]], pAdjustMethod="BH", keyType="ENTREZID",OrgDb="org.Dr.eg.db",readable = T, ont="BP", pvalueCutoff = 0.05)

ewp_2 <- clusterProfiler::enrichGO(gene=differ_4_2[[2]], pAdjustMethod="BH", keyType="ENTREZID",OrgDb="org.Dr.eg.db",readable = T, ont="BP", pvalueCutoff = 0.05)

ewp_3 <- clusterProfiler::enrichGO(gene=differ_4_3[[2]], pAdjustMethod="BH", keyType="ENTREZID",OrgDb="org.Dr.eg.db",readable = T, ont="BP", pvalueCutoff = 0.05)

ewp_4 <- clusterProfiler::enrichGO( gene=differ_4_4[[2]], pAdjustMethod="BH", keyType="ENTREZID",OrgDb="org.Dr.eg.db",readable = T, ont="BP", pvalueCutoff = 0.05)

ewp_5 <-clusterProfiler::enrichGO(gene=differ_4_5[[2]], pAdjustMethod="BH", keyType="ENTREZID",OrgDb="org.Dr.eg.db",readable = T, ont="BP", pvalueCutoff = 0.05)

ewp_6 <-clusterProfiler::enrichGO(gene=differ_4_6[[2]], pAdjustMethod="BH", keyType="ENTREZID",OrgDb="org.Dr.eg.db",readable = T, ont="BP", pvalueCutoff = 0.05)

ewp_1@result$FDR2<- -log10(ewp_1@result$p.adjust )
GO04_1<- ggplot(ewp_1@result[1:10,], aes(x=reorder(Description, FDR2),FDR2,  fill=FDR2)) +    geom_bar(stat = "identity") +    coord_flip() +    scale_fill_continuous(low="blue", high="deeppink") +    labs(x = "", y = "", fill = "FDR2",title = paste(PROJECT,"enrichGO_TOP10_BP_c1",sep="")) + theme_classic() +    theme(axis.text=element_text(size=11))  + ggtitle(paste(PROJECT,"enrichGO_TOP5_BP_c1",sep=""))
 
ewp_2@result$FDR2<- -log10(ewp_2@result$p.adjust )
GO04_2<- ggplot(ewp_2@result[1:10,], aes(x=reorder(Description, FDR2),FDR2,  fill=FDR2)) +    geom_bar(stat = "identity") +    coord_flip() +    scale_fill_continuous(low="blue", high="deeppink") +    labs(x = "", y = "", fill = "FDR2",title = paste(PROJECT,TRY,"enrichGO_c2")) + theme_classic() +    theme(axis.text=element_text(size=11))  + ggtitle(paste(PROJECT,"enrichGO_TOP5_BP_c2",sep=""))

ewp_3@result$FDR2<- -log10(ewp_3@result$p.adjust )
GO04_3<- ggplot(ewp_3@result[1:10,], aes(x=reorder(Description, FDR2),FDR2,  fill=FDR2)) +    geom_bar(stat = "identity") +    coord_flip() +    scale_fill_continuous(low="blue", high="deeppink") +    labs(x = "", y = "", fill = "FDR2",title = paste(PROJECT,TRY,"enrichGO_c3")) + theme_classic() +    theme(axis.text=element_text(size=11))  + ggtitle(paste(PROJECT,"enrichGO_TOP5_BP_c3",sep=""))

ewp_4@result$FDR2<- -log10(ewp_4@result$p.adjust )
GO04_4<- ggplot(ewp_4@result[1:10,], aes(x=reorder(Description, FDR2),FDR2,  fill=FDR2)) +    geom_bar(stat = "identity") +  coord_flip() +    scale_fill_continuous(low="blue", high="deeppink") +    labs(x = "", y = "", fill = "FDR2",title = paste(PROJECT,TRY,"enrichGO_c4")) + theme_classic() +    theme(axis.text=element_text(size=11))  + ggtitle(paste(PROJECT,"enrichGO_TOP5_BP_c4",sep=""))

ewp_5@result$FDR2<- -log10(ewp_5@result$p.adjust )
GO04_5<- ggplot(ewp_5@result[1:10,], aes(x=reorder(Description, FDR2),FDR2,  fill=FDR2)) +    geom_bar(stat = "identity") +  coord_flip() +    scale_fill_continuous(low="blue", high="deeppink") +    labs(x = "", y = "", fill = "FDR2",title = paste(PROJECT,TRY,"enrichGO_c5")) + theme_classic() +    theme(axis.text=element_text(size=11))  + ggtitle(paste(PROJECT,"enrichGO_TOP5_BP_c5",sep=""))

ewp_6@result$FDR2<- -log10(ewp_6@result$p.adjust )
GO04_6<- ggplot(ewp_6@result[1:10,], aes(x=reorder(Description, FDR2),FDR2,  fill=FDR2)) +    geom_bar(stat = "identity") +    coord_flip() +    scale_fill_continuous(low="blue", high="deeppink") +    labs(x = "", y = "", fill = "FDR2",title = paste(PROJECT,TRY,"enrichGO_c5")) + theme_classic() +    theme(axis.text=element_text(size=11))  + ggtitle(paste(PROJECT,"enrichGO_TOP5_BP_c6",sep=""))
```


## Print GO to check the related Biological function to the clusters
```{r,fig.dim = c(9, 9),message=FALSE,class.source="bg-info",warning=FALSE}
print(GO001)
print(GO00b1)
```

## write the the GO into Tables
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
pdf(file=paste(PORT3,"004_forFig2G_",PROJECT, "GO",".pdf",sep=""), width=10, height=7)  
print(GO001)
print(GO00b1)
print(GO04_1)
print(GO04_2)
print(GO04_3)
print(GO04_4)
print(GO04_5)
print(GO04_6)
dev.off()
NAME_GO<- c("ID", "pvalue","GeneRatio", "p.adjust","Description","geneID")
egoFDR2<- subset(formula_resddBGI11FC1_25_1_5_bonfe@compareClusterResult, ID=" ", select=NAME_GO)
egoFDR3_1<- subset(ewp_1@result, select=NAME_GO)
egoFDR3_2<- subset(ewp_2@result, select=NAME_GO)
egoFDR3_3<- subset(ewp_3@result, select=NAME_GO)
egoFDR3_4<- subset(ewp_4@result, select=NAME_GO)
egoFDR3_5<- subset(ewp_5@result, select=NAME_GO)
egoFDR3_6<- subset(ewp_6@result,select=NAME_GO)
egoFDR3_1$REG<-"c1"
egoFDR3_2$REG<- "c2"
egoFDR3_3$REG<- "c3"
egoFDR3_4$REG<- "c4"
egoFDR3_5$REG<- "c5"
egoFDR3_6$REG<- "c6"
egoFDR3diff <- rbind(egoFDR3_1,egoFDR3_2,egoFDR3_3,egoFDR3_4,egoFDR3_5,egoFDR3_6)
#write.xlsx(egoFDR2,file=paste(PTHA,"004_forFig2G_",PROJECT,'FIGURES_GOs_bonferro_MATRIX_2.xlsx',sep=""),overwrite = T)
#write.table(egoFDR2,file=paste(PTHA,"004_forFig2G_",PROJECT,"FIGURES_GOs_bonferro_MATRIX_2.txt",sep=""),sep="\t",row.names = F,col.names=T,dec=".",quote = F)
#write.xlsx(formula_resddBGI11FC1_25_1_5_FDR@compareClusterResult,file=paste(PTHA,PROJECT,'FIGURES_GOs_FDR_MATRIX.xlsx',sep=""),overwrite = T)
#write.xlsx(egoFDR3diff,file=paste(PTHA,"004_forFig2G_",PROJECT,"to_select_GO.xls",sep=""),overwrite = T)
#write.table(egoFDR3diff,file=paste(PTHA,"004_forFig2G_",PROJECT,"_FDR_MATRIX_forREVIGO_moreRPKM_Q1.txt",sep=""),sep="\t",row.names = F,col.names=T,dec=".",quote = F)
```
##### Barplots for the region and Peak size  ###########
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE}
NAME44<- data.frame(colnames(MAT11))
mat_adD12 <- subset(MAT11,CLU2=="01PROM(<2kb)" | CLU2=="02Genebody(>2kb)"| CLU2=="03Interg", select=c("PEAK_ID2","Clu","ID","CLU","CLU2","COR","COM"))
NAME_CORR<- c("PEAK_ID2","Clu","ID","CLU","CLU2","COR","COM")
colnames(mat_adD12)<- NAME_CORR
MA6<- data.frame(table(mat_adD12$CLU,mat_adD12$ID,mat_adD12$Clu))
MA7<- data.frame(table(mat_adD12$CLU2,mat_adD12$ID,mat_adD12$Clu))
MA9<- data.frame(table(mat_adD12$COM,mat_adD12$ID,mat_adD12$Clu))
MA10<- data.frame(table(mat_adD12$COM,mat_adD12$ID,mat_adD12$Clu,mat_adD12$CLU))
MA10$Var5<- paste(MA10$Var4,MA10$Var1,sep="_")
CO22_COR2<-brewer.pal(n = 8, name = "Set2")
```


```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE,fig.keep='none'}
###option fill is giving the result in percentage and stack in total values
SI=10
XL="dpci"
YL="Peak wide (kb)"
 PROM="gold"
 GENBODY="#4393C3"
 INTER="brown"
 NF="#31a354"
 CO23 <-  c("gold","darkblue","brown", "gray","darkblue", "black")

 CO22 <-  c(PROM,GENBODY,INTER)
    BROAD="#4d004b"
    MEDIUM="#8c6bb1"
    NARROW="#9ebcda"
    CO22P <-  c(BROAD,NARROW)
 NAME33b<-c("00", "01", "04", "14", "45") 
 p1 <- ggplot(MA6, aes(factor(Var2), Freq, fill = factor(Var1))) +     geom_bar(position = "fill",stat = "identity") +      guides(fill = guide_legend(reverse = F)) +    labs(fill = "Peak group (kb)")+  theme_bw()
  p6 <- p1 +   scale_fill_manual(values =CO22P)+  scale_y_continuous(breaks = c(0,0.25,0.75)) +  theme(strip.placement = "inside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.text.y=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.x = element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.y=element_text(angle = 90,color=BASIC_COL, size=SI, face="bold"),panel.border = element_rect(colour = BASIC_COL,  size=2)) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL2)+ scale_x_discrete(labels=NAME33b)+
     facet_wrap(~Var3, ncol = 8,strip.position="top")+   theme_bw()
  
  p1b <- ggplot(MA6, aes(factor(Var2), Freq, fill = factor(Var1))) +  geom_bar(position = "stack",stat = "identity") +    guides(fill = guide_legend(reverse = F)) +          labs(fill = "Peak group (kb)")+ theme_bw()
  (p6b <- p1b +   scale_fill_manual(values =CO22P) + theme(strip.placement = "inside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.text.y=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.x = element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.y=element_text(angle = 90,color=BASIC_COL, size=SI, face="bold"),panel.border = element_rect(colour = BASIC_COL,  size=2)) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL2)+ scale_x_discrete(labels=NAME33b)+  facet_wrap(~Var3, ncol = 8,strip.position="top"))+     theme_bw()

  p2 <- ggplot(MA7, aes(factor(Var2), Freq, fill = factor(Var1))) +geom_bar(position = "fill",stat = "identity") +  guides(fill = guide_legend(reverse = F)) +
          labs(fill = "Peak group (kb)")+   theme_bw()
  p6c <- p2 +  scale_fill_manual(values =CO22)+  scale_y_continuous(breaks = c(0,0.25,0.75)) +  theme(strip.placement = "inside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.text.y=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.x = element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.y=element_text(angle = 90,color=BASIC_COL, size=SI, face="bold"),panel.border = element_rect(colour = BASIC_COL,  size=2)) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL)+ scale_x_discrete(labels=NAME33b)+ facet_wrap(~Var3, ncol = 8,strip.position="top")+   theme_bw()

  p2d <- ggplot(MA7, aes(factor(Var2), Freq, fill = factor(Var1))) +geom_bar(position = "stack",stat = "identity") +guides(fill = guide_legend(reverse = F)) +  labs(fill = "Genomic region")+   theme_bw()
  
  (p6d <- p2d +  scale_fill_manual(values =CO22) +  theme(strip.placement = "inside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.text.y=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.x = element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.y=element_text(angle = 90,color=BASIC_COL, size=SI, face="bold"),panel.border = element_rect(colour = BASIC_COL,  size=2)) + labs(title=paste(PROJECT,sep="" ) , x=XL, y = YL2)+ scale_x_discrete(labels=NAME33b)+ facet_wrap(~Var3, ncol = 8,strip.position="top"))+  theme_bw()

    
pdf(file=paste(PTHA,"005_forFigS2_",PROJECT,"_by_CLUS",".pdf", sep=""),width=10, height=5)  
arrange0 <- ggarrange(p6,p6b,     ncol = 1,nrow =2,common.legend = T, align = c("none"),legend="top")
arrange01 <- ggarrange(p6c ,p6d,    ncol = 1,nrow =2,common.legend = T, align = c("hv"),legend="top")


print(arrange0)
print(arrange01)
dev.off()
```

### Correlation of peak by cluster by genome area (Promoter, Genebody and Intergenic) ###

```{r,fig.dim = c(7, 8),class.source="bg-info"}
print(arrange0)
print(arrange01) 
```

### Barplot of the peak broadness by Clusters ###
```{r,message=FALSE,class.source="bg-info",warning=FALSE,echo=FALSE,fig.keep='none'}
YL3<- "Distr of peaks (%)"
SI=11
p9 <- ggplot(MA9, aes(factor(Var2), Freq, fill = factor(Var1))) +  geom_bar(position = "fill",stat = "identity") +  guides(fill = guide_legend(reverse = F)) +labs(fill = "corr")+ theme_bw() +   scale_fill_manual(values =CO22_COR)+  scale_y_continuous(breaks = c(0,0.25,0.75)) +  theme(strip.placement = "inside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.text.y=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.x = element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.y=element_text(angle = 90,color=BASIC_COL, size=SI, face="bold"),panel.border = element_rect(colour = BASIC_COL,  size=2)) + labs(title=paste("004_","FigS2E_", PROJECT,"CORR_SEPA_by_CORR",sep="" ) , x=XL, y = YL3)+ scale_x_discrete(labels=NAME33b)+     facet_wrap(~Var3, ncol = 8,strip.position="top")

p91 <- ggplot(MA9, aes(factor(Var2), Freq, fill = factor(Var1))) +geom_bar(position = "stack",stat = "identity") +  guides(fill = guide_legend(reverse = F)) +labs(fill = "corr")+     theme_bw() +   scale_fill_manual(values =CO22_COR)+  scale_y_continuous(breaks = c(0,1000,2200)) +  theme(strip.placement = "inside",strip.text.y.left = element_text(angle=0),axis.text.x=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.text.y=element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.x = element_text(angle = 0,color=BASIC_COL, size=SI, face="bold"),axis.title.y=element_text(angle = 90,color=BASIC_COL, size=SI, face="bold"),panel.border = element_rect(colour = BASIC_COL,  size=2)) + labs(title=paste("004_","FigS2E_", PROJECT,"CORR_SEPA_by_CORR",sep="" ) , x=XL, y = YL2)+ scale_x_discrete(labels=NAME33b)+     facet_wrap(~Var3, ncol = 8,strip.position="top")

{pdf(file=paste("006_","FigS2E_", PROJECT,"CORR_SEPA_by_CORR",".pdf", sep=""),width=10, height=7)  
arrange11 <- ggarrange(p9,p91,  ncol = 1,nrow =2,common.legend = T, align = c("hv"),legend="top")
print(arrange11)            
dev.off()
     }
```

### Barplot of the peak broadness by Clusters ###

```{r,fig.dim = c(7, 8),class.source="bg-info"}
print(arrange11)  
```
